dicttonx: no longer accepts pathlib.Path

This works fine in relase but breaks in the main branch:

```python
_______________________________________________________________________________________ test_execution_with_different_output_files ________________________________________________________________________________________

tmp_path = PosixPath('/tmp/pytest-of-denolf/pytest-0/test_execution_with_different_0')

    @pytest.mark.skipif(not is_silx_compatible(), reason="Silx is too old")
    def test_execution_with_different_output_files(tmp_path):
        q = numpy.linspace(0.01, 10, 100, dtype=numpy.float64)
        input_data = 2 * numpy.ones((2, 100), dtype=numpy.float32)
        background_data = numpy.ones((1, 100), dtype=numpy.float32)
        background_factor = 0.1
    
        # Create input file
        input_filename = tmp_path / "input.h5"
        input_nxdata = {
            **NXDATA_METADATA,
            "q": q,
            "q@units": "A^-1",
            "intensity": input_data,
            "points": [0, 1],
        }
>       dicttonx(input_nxdata, input_filename, "/entry/process/integrated")

src/ewoksxrpd/tests/test_subtract_pattern.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

treedict = {'@NX_class': 'NXdata', '@axes': ['.', 'q'], '@interpretation': 'spectrum', '@signal': 'intensity', ...}, h5file = PosixPath('/tmp/pytest-of-denolf/pytest-0/test_execution_with_different_0/input.h5')
h5path = '/entry/process/integrated/', add_nx_class = None, kw = {}

    def dicttonx(treedict, h5file, h5path="/", add_nx_class=None, **kw):
        """
        Write a nested dictionary to a HDF5 file, using string keys as member names.
        The NeXus convention is used to identify attributes with ``"@"`` character,
        therefore the dataset_names should not contain ``"@"``.
    
        Similarly, links are identified by keys starting with the ``">"`` character.
        The corresponding value can be a soft link, external link or virtual dataset.
    
        :param treedict: Nested dictionary/tree structure with strings as keys
             and array-like objects as leafs. The ``"/"`` character can be used
             to define sub tree. The ``"@"`` character is used to write attributes.
             The ``">"`` prefix is used to define links.
        :param add_nx_class: Add "NX_class" attribute when missing. By default it
            is ``True`` when ``update_mode`` is ``"add"`` or ``None``.
    
        The named parameters are passed to dicttoh5.
    
        Example::
    
            import numpy
            from silx.io.dictdump import dicttonx
    
            gauss = {
                "entry": {
                    "title": "A plot of a gaussian",
                    "instrument": {
                        "@NX_class": "NXinstrument",
                        "positioners": {
                            "@NX_class": "NXCollection",
                            "x": numpy.arange(0,1.1,.1)
                        }
                    }
                    "plot": {
                        "y": numpy.array([0.08, 0.19, 0.39, 0.66, 0.9, 1.,
                                      0.9, 0.66, 0.39, 0.19, 0.08]),
                        ">x": "../instrument/positioners/x",
                        "@signal": "y",
                        "@axes": "x",
                        "@NX_class": "NXdata",
                        "title": "Gauss Plot",
                     },
                     "@NX_class": "NXentry",
                     "default":"plot",
                }
                "@NX_class": "NXroot",
                "@default": "entry",
            }
    
            dicttonx(gauss,"test.h5")
        """
        h5file, h5path = _normalize_h5_path(h5file, h5path)
        if isinstance(h5file, str):
            file_path = h5file
        else:
>           file_path = h5file.filename
                        ^^^^^^^^^^^^^^^
E           AttributeError: 'PosixPath' object has no attribute 'filename'

../silx/src/silx/io/dictdump.py:788: AttributeError
```

--- a/src/silx/io/dictdump.py
+++ b/src/silx/io/dictdump.py
@@ -782,10 +782,10 @@ def dicttonx(treedict, h5file, h5path="/", add_nx_class=None, **kw):
         dicttonx(gauss,"test.h5")
     """
     h5file, h5path = _normalize_h5_path(h5file, h5path)
-    if isinstance(h5file, str):
-        file_path = h5file
-    else:
+    if is_h5_file_like(h5file):
         file_path = h5file.filename
+    else:
+        file_path = str(h5file)
     parents = tuple(p for p in h5path.split("/") if p)
     if add_nx_class is None:
         add_nx_class = kw.get("update_mode", None) in (None, "add")

